### DISCUSSION TITLE
How can I know how many frames rejected for stitching algorithm

### DISCUSSION THREAD

**Discussion by thilnguy** [OpenStitching/stitching#194] [Q&A] [answered]
Hello,
Sometimes, I get the message with the content like this "not all frames can be stitched". From the verbose log, how can I know how many frames are rejected? How can I know the angle of panorama it can cover if not all images stitched?
Thank you

**Comment by lukasalexanderweber** [comment]
You can check out the matches graph to see which images are rejected

**Comment by thilnguy** [comment]
Thanks a lot for your quick reply.
Here is my graph: 
graph matches_graph{
"1" -- "6"[label="Nm=49, Ni=46, C=2.02643"];
"2" -- "5"[label="Nm=91, Ni=51, C=1.44476"];
"2" -- "10"[label="Nm=29, Ni=25, C=1.49701"];
"2" -- "11"[label="Nm=20, Ni=12, C=0.857143"];
"2" -- "14"[label="Nm=74, Ni=45, C=1.49007"];
"3" -- "4"[label="Nm=113, Ni=80, C=1.90931"];
"3" -- "25"[label="Nm=154, Ni=128, C=2.36162"];
"3" -- "26"[label="Nm=68, Ni=59, C=2.07746"];
"3" -- "29"[label="Nm=39, Ni=29, C=1.47208"];
"4" -- "30"[label="Nm=21, Ni=19, C=1.32867"];
"5" -- "7"[label="Nm=51, Ni=50, C=2.14592"];
"7" -- "27"[label="Nm=138, Ni=119, C=2.40891"];
"7" -- "32"[label="Nm=35, Ni=23, C=1.24324"];
"8" -- "18"[label="Nm=25, Ni=23, C=1.48387"];
"11" -- "16"[label="Nm=124, Ni=109, C=2.4115"];
"11" -- "17"[label="Nm=21, Ni=11, C=0.769231"];
"12" -- "13"[label="Nm=34, Ni=33, C=1.81319"];
"12" -- "15"[label="Nm=26, Ni=26, C=1.64557"];
"12" -- "18"[label="Nm=14, Ni=13, C=1.06557"];
"12" -- "21"[label="Nm=14, Ni=13, C=1.06557"];
"16" -- "20"[label="Nm=18, Ni=15, C=1.1194"];
"19" -- "22"[label="Nm=144, Ni=136, C=2.65625"];
"19" -- "23"[label="Nm=126, Ni=103, C=2.24891"];
"19" -- "24"[label="Nm=31, Ni=29, C=1.6763"];
"19" -- "28"[label="Nm=14, Ni=11, C=0.901639"];
"19" -- "32"[label="Nm=49, Ni=42, C=1.85022"];
"20" -- "21"[label="Nm=7, Ni=7, C=0.693069"];
"24" -- "26"[label="Nm=7, Ni=5, C=0.49505"];
"24" -- "31"[label="Nm=87, Ni=71, C=2.08211"];
"9";
}
I am not so sure that I can check which images are rejected. For example, 1->6, then? both 1 and 6 are rejected?
In your tutorial https://github.com/OpenStitching/stitching_tutorial/blob/master/Stitching%20Tutorial.ipynb,
 it is quite clear when image1 connected to imaged 2, then connected to image3.
 That would be great if you can point me how it works in this graph.
Thank you

**Reply by thilnguy** [reply]
And I have second question. In production, i will run stitch() instead of stitch_verbose(). Is there any way to check the rejected images, please?

**Reply by lukasalexanderweber** [reply]
You can create your custom stitcher class where you can include any reporting you want

**Reply by lukasalexanderweber** [reply, marked as answer]
[here](https://dreampuf.github.io/GraphvizOnline/#%20graph%20matches_graph%7B%0A%221%22%20--%20%226%22%5Blabel%3D%22Nm%3D49%2C%20Ni%3D46%2C%20C%3D2.02643%22%5D%3B%0A%222%22%20--%20%225%22%5Blabel%3D%22Nm%3D91%2C%20Ni%3D51%2C%20C%3D1.44476%22%5D%3B%0A%222%22%20--%20%2210%22%5Blabel%3D%22Nm%3D29%2C%20Ni%3D25%2C%20C%3D1.49701%22%5D%3B%0A%222%22%20--%20%2211%22%5Blabel%3D%22Nm%3D20%2C%20Ni%3D12%2C%20C%3D0.857143%22%5D%3B%0A%222%22%20--%20%2214%22%5Blabel%3D%22Nm%3D74%2C%20Ni%3D45%2C%20C%3D1.49007%22%5D%3B%0A%223%22%20--%20%224%22%5Blabel%3D%22Nm%3D113%2C%20Ni%3D80%2C%20C%3D1.90931%22%5D%3B%0A%223%22%20--%20%2225%22%5Blabel%3D%22Nm%3D154%2C%20Ni%3D128%2C%20C%3D2.36162%22%5D%3B%0A%223%22%20--%20%2226%22%5Blabel%3D%22Nm%3D68%2C%20Ni%3D59%2C%20C%3D2.07746%22%5D%3B%0A%223%22%20--%20%2229%22%5Blabel%3D%22Nm%3D39%2C%20Ni%3D29%2C%20C%3D1.47208%22%5D%3B%0A%224%22%20--%20%2230%22%5Blabel%3D%22Nm%3D21%2C%20Ni%3D19%2C%20C%3D1.32867%22%5D%3B%0A%225%22%20--%20%227%22%5Blabel%3D%22Nm%3D51%2C%20Ni%3D50%2C%20C%3D2.14592%22%5D%3B%0A%227%22%20--%20%2227%22%5Blabel%3D%22Nm%3D138%2C%20Ni%3D119%2C%20C%3D2.40891%22%5D%3B%0A%227%22%20--%20%2232%22%5Blabel%3D%22Nm%3D35%2C%20Ni%3D23%2C%20C%3D1.24324%22%5D%3B%0A%228%22%20--%20%2218%22%5Blabel%3D%22Nm%3D25%2C%20Ni%3D23%2C%20C%3D1.48387%22%5D%3B%0A%2211%22%20--%20%2216%22%5Blabel%3D%22Nm%3D124%2C%20Ni%3D109%2C%20C%3D2.4115%22%5D%3B%0A%2211%22%20--%20%2217%22%5Blabel%3D%22Nm%3D21%2C%20Ni%3D11%2C%20C%3D0.769231%22%5D%3B%0A%2212%22%20--%20%2213%22%5Blabel%3D%22Nm%3D34%2C%20Ni%3D33%2C%20C%3D1.81319%22%5D%3B%0A%2212%22%20--%20%2215%22%5Blabel%3D%22Nm%3D26%2C%20Ni%3D26%2C%20C%3D1.64557%22%5D%3B%0A%2212%22%20--%20%2218%22%5Blabel%3D%22Nm%3D14%2C%20Ni%3D13%2C%20C%3D1.06557%22%5D%3B%0A%2212%22%20--%20%2221%22%5Blabel%3D%22Nm%3D14%2C%20Ni%3D13%2C%20C%3D1.06557%22%5D%3B%0A%2216%22%20--%20%2220%22%5Blabel%3D%22Nm%3D18%2C%20Ni%3D15%2C%20C%3D1.1194%22%5D%3B%0A%2219%22%20--%20%2222%22%5Blabel%3D%22Nm%3D144%2C%20Ni%3D136%2C%20C%3D2.65625%22%5D%3B%0A%2219%22%20--%20%2223%22%5Blabel%3D%22Nm%3D126%2C%20Ni%3D103%2C%20C%3D2.24891%22%5D%3B%0A%2219%22%20--%20%2224%22%5Blabel%3D%22Nm%3D31%2C%20Ni%3D29%2C%20C%3D1.6763%22%5D%3B%0A%2219%22%20--%20%2228%22%5Blabel%3D%22Nm%3D14%2C%20Ni%3D11%2C%20C%3D0.901639%22%5D%3B%0A%2219%22%20--%20%2232%22%5Blabel%3D%22Nm%3D49%2C%20Ni%3D42%2C%20C%3D1.85022%22%5D%3B%0A%2220%22%20--%20%2221%22%5Blabel%3D%22Nm%3D7%2C%20Ni%3D7%2C%20C%3D0.693069%22%5D%3B%0A%2224%22%20--%20%2226%22%5Blabel%3D%22Nm%3D7%2C%20Ni%3D5%2C%20C%3D0.49505%22%5D%3B%0A%2224%22%20--%20%2231%22%5Blabel%3D%22Nm%3D87%2C%20Ni%3D71%2C%20C%3D2.08211%22%5D%3B%0A%229%22%3B%0A%7D) you see your graph

**Reply by thilnguy** [reply]
Thanks a lot. !!!!
